zerver/tests/test_events.py

```diff
@@ -523,6 +526,8 @@ def do_test(self, action: Callable[[], Any], event_types: Optional[List[str]]=No
             if before != after:
                 raise AssertionError('Test is invalid--state actually does change here.')
 
+        self.user_profile = get_user_profile_by_id(self.user_profile.id)
+
```

Comment:
This is an interesting issue!  Arguably, the bug here is that we use `realm = get_realm("zulip")` in that test, rather than just passing `self.user_profile.realm` to `do_change_icon_source` in the following tests -- because that's what caused the local `realm` object that we updated in our `do_*` function to be different from what we accessed here.

What do you think about instead fixing this by removing all uses of `get_realm` in `test_events.py`, and adding a lint rule banning that in this file to prevent further recurrences of that sort of bug?  

I suppose possibly we both merge this and make that change (since such calls to `get_realm`  are all unnecessary).